app/javascript/components/track/IterationSummary.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect, useState, useRef } from 'react'
+import { TestRunStatus } from '../student/Editor'
+import { fromNow } from '../../utils/time'
+import { SubmissionMethodIcon } from './iteration-summary/SubmissionMethodIcon'
+import { TestsStatusSummary } from './iteration-summary/TestsStatusSummary'
+import { AnalysisStatusSummary } from './iteration-summary/AnalysisStatusSummary'
+import { IterationsChannel } from '../../channels/iterationsChannel'
+
+export type Iteration = {
+  id: number
+  idx: number
+  submissionMethod: SubmissionMethod
+  createdAt: Date
+  testsStatus: TestRunStatus
+  representationStatus: RepresentationStatus
+  analysisStatus: AnalysisStatus
+}
+
+export enum SubmissionMethod {
+  CLI = 'cli',
+  API = 'api',
+}
+
+export enum RepresentationStatus {
+  NOT_QUEUED = 'not_queued',
+  QUEUED = 'queued',
+  APPROVED = 'approved',
+  DISAPPROVED = 'disapproved',
+  INCONCLUSIVE = 'inconclusive',
+  EXCEPTIONED = 'exceptioned',
+  CANCELLED = 'cancelled',
+}
+
+export enum AnalysisStatus {
+  NOT_QUEUED = 'not_queued',
+  QUEUED = 'queued',
+  APPROVED = 'approved',
+  DISAPPROVED = 'disapproved',
+  INCONCLUSIVE = 'inconclusive',
+  EXCEPTIONED = 'exceptioned',
+  CANCELLED = 'cancelled',
+}
+
+export function IterationSummary(props: { iteration: Iteration }) {
+  const [iteration, setIteration] = useState(props.iteration)
+  const submissionMethodLabels = {
+    [SubmissionMethod.CLI]: 'CLI',
+    [SubmissionMethod.API]: 'API',
+  }
+  const channel = useRef<IterationsChannel | undefined>()
+
+  useEffect(() => {
+    channel.current = new IterationsChannel(iteration, (iteration) => {
+      setIteration(iteration)
+    })
+
+    return () => {
+      channel.current?.disconnect()
+    }
+  }, [iteration])
```

Comment:
This can lead to a subtle bug. By the time line 58 runs, `channel.current` may be a _different object.

```suggestion
  useEffect(() => {
    const iteractionChannel = new IterationsChannel(iteration, setIteration)
    
    // This will *not* trigger a re-render of this component. If you need that,
    // use `const [channel, setChannel] = useState()` instead.
    channel.current = iteractionChannel
     
    return () => {
      iteractionChannel.disconnect()
    }
  }, [channel, iteration, setIteration])
```

Reply:
Nice catch!

Reply:
Thanks!